Formula/l/llama-cpp.rb

```diff
@@ -0,0 +1,32 @@
+class LlamaCpp < Formula
+  desc "LLM inference in C/C++"
+  homepage "https://github.com/ggerganov/llama.cpp"
+  url "https://github.com/ggerganov/llama.cpp/archive/refs/tags/b2715.tar.gz"
+  sha256 "8b1831b605ee4c564ff84e648dabdc11fe3d77dcb6b1d7e4d97a14ea991755f2"
+  license "MIT"
+  head "https://github.com/ggerganov/llama.cpp.git", branch: "master"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+
+    if OS.mac?
+      bin.install "build/bin/default.metallib"
+      bin.install_symlink lib/"default.metallib"
+    end
+
+    bin.install "build/bin/finetune" => "llama-cpp-finetune"
+    bin.install "build/bin/main" => "llama-cpp"
+    bin.install "build/bin/perplexity" => "llama-cpp-perplexity"
+    bin.install "build/bin/quantize" => "llama-cpp-quantize"
+    bin.install "build/bin/server" => "llama-cpp-server"
+    pkgshare.install "build/bin/test-sampling"
```

Comment:
Is there no `cmake --install`?

Reply:
cmake --install installs every example built during compilation which is unnecessary. There is no way to compile just one example in llama.cpp so had to install separate binaries like above since these are the important ones. Rest of the examples are not required to be installed since above ones are the ones used practically in real world.

Reply:
Using --install doesn't prepend the "llama-cpp" prefix in the binaries installed so it will be very confusing for anyone by the way. Suggest we go with the most recent changes here. Also, llama.cpp is inherently a library and examples folder in the repo was just a way for ggerganov to showcase its usage and abilities. But some of the examples are widely used in LLMs world like main, quantize, and finetune and to keep it neat I had to prefix it with "llama-cpp"